### PR TITLE
Fix ThirdPartyDepsAnalyzer

### PR DESCRIPTION
**Pull request by val-fed**

### CODE REVIEW

File: tools/build_file_generator/third_party_deps/BUILD [hunk 1]
```diff
@@ -21,42 +21,41 @@ checkstyle_test(
     target = ":third_party_deps_analyzer",
 )
 
-# '%START%' and '%END%' are markers for 'third_party_deps_updater.sh' script.
-# Don't delete it while 'third_party_deps_updater.sh' script is used.
+# '%START%' and '%END%' are markers for 'third_party_deps_updater.sh' script and ThirdPartyDepsAnalyzer class.
+# Don't delete it while 'third_party_deps_updater.sh' script and ThirdPartyDepsAnalyzer class are used.
 THIRD_PARTY_DEPS_GROUP = [
     # %START%
```
Comment (oferb) [inline tools/build_file_generator/third_party_deps/BUILD]: Maybe we should rename to `THIRD_PARTY_DEPS_START`, and similarly for `END`?

File: tools/build_file_generator/third_party_deps/BUILD [hunk 2]
```diff
@@ -21,42 +21,41 @@ checkstyle_test(
...
     "//third_party/maven/com/google/dagger:dagger_compiler",
     "//third_party/maven/com/google/errorprone:error_prone_ant",
     "//third_party/maven/com/google/firebase:firebase_admin",
-    "//third_party/maven/com/google/flogger",
+    "//third_party/maven/com/google/flogger:flogger",
     "//third_party/maven/com/google/flogger:flogger_system_backend",
     "//third_party/maven/com/google/googlejavaformat:google_java_format",
-    "//third_party/maven/com/google/guava",
-    "//third_party/maven/com/google/jimfs",
+    "//third_party/maven/com/google/guava:guava",
```
Comment (oferb) [inline tools/build_file_generator/third_party_deps/BUILD]: There's no need to repeat the target if the folder name is the same:
`//third_party/maven/com/google/guava:guava`
Is the same as:
`//third_party/maven/com/google/guava`

Is it easy to shorten them?

File: tools/build_file_generator/third_party_deps/BUILD (hunk 1 above)
Comment (val-fed) [reply tools/build_file_generator/third_party_deps/BUILD]: Done.

File: tools/build_file_generator/third_party_deps/BUILD (hunk 2 above)
Comment (val-fed) [reply tools/build_file_generator/third_party_deps/BUILD]: Added logic to shorten it to the script.

**Pull request merged**